src/flask_cognito_lib/decorators.py

```diff
@@ -32,6 +35,42 @@ def remove_from_session(keys: Iterable[str]):
                 session.pop(key)
 
 
+def store_tokens(tokens: CognitoTokenResponse, nonce: Optional[str] = None) -> None:
+    """Store the tokens in the session"""
+    # validate the JWT and get the claims
+    claims = cognito_auth.verify_access_token(
+        token=tokens.access_token,
+        leeway=cfg.cognito_expiration_leeway,
+    )
+    session.update({"claims": claims})
+
+    # Grab the user info from the user endpoint and store in the session
+    if tokens.id_token is not None:
+        user_info = cognito_auth.verify_id_token(
+            token=tokens.id_token,
+            nonce=nonce,
+            leeway=cfg.cognito_expiration_leeway,
+        )
+        session.update({"user_info": user_info})
+
+    # Grab the `refresh_token` and store in the session
+    if tokens.refresh_token is not None:
+        session.update({"refresh_token": tokens.refresh_token})
```

Comment:
I think for security, we probably want to store the refresh token in a cookie rather than in the session

Reply:
I'm still a little nervous about storing the refresh token client side, as the refresh token flow doesn't have any of the added security bits of the initial auth grant flow (like nonce, PKCE etc.) and I think CSRF is potentially an issue if the refresh tokens get leaked. Maybe we want to consider whether a server side session is better option. What is the use case for this flow in your application?

Reply:
Are Flask sessions so bad? :sweat_smile:

I thought even for client-side session `SECRET_KEY` is used to encrypt the contents, in addition developers should consider using `HttpOnly` and `Secure` with appropriate `Same-Site` policy.
All the above should make `session` a fairly secure storage.

Though, if we want to enforce `HttpOnly` and `Secure` on our side - sure can be a separate cookie.

> What is the use case for this flow in your application?
I have a React front end, that ~5 min before expiration shows user a modal with the button to prolong the session. 
If user clicks it - refresh flow happens. 
If not - user will be kicked-off. Expiration time for cookies, `access_token`, and `refresh_token` are aligned.

Reply:
> Are Flask sessions so bad? 😅

They're not a place to store secrets really. The [secret key is used to cryptographically sign](https://flask.palletsprojects.com/en/3.0.x/quickstart/#sessions) the contents of the session (i.e. it can't be modified), but it's only base64 encoded (maybe zlib compressed, if large), so the [contents of the session](https://blog.miguelgrinberg.com/post/how-secure-is-the-flask-user-session) can still be extracted pretty easily, hence it's definitely not the place we want to put the refresh token. 

So I'm still a bit on the fence about the refresh token. When it comes to access token we have a bunch of additional security because the flow requires the client id / client secret, nonce, and PKCE are used, but the refresh token flow doesn't require any of this - hence why I'm thinking server side storage might be needed.

Reply:
Ok, thank you for the feedback.

I'll do more research to see what can be done here.

Reply:
@mblackgeo I put potential solution in the https://github.com/mblackgeo/flask-cognito-lib/pull/39/commits/8665ecca73074bbaced85cea75dac1b9ad332d00

In short it is:
1. Added toggle for `Refresh Flow` via config
2. Store the refresh token in HTTP-only secure cookie
3. Added toggle to `Encrypt` the refresh token server-side

If that looks good will work on tests.